torch_xla/csrc/aten_xla_bridge.cpp

```diff
@@ -128,9 +150,11 @@ absl::Status ReplaceXlaTensor(const at::Tensor& tensor,
 absl::Status ReplaceXlaTensor(const std::vector<at::Tensor>& tensors,
                               const std::vector<XLATensorPtr> new_xla_tensors) {
   if (tensors.size() != new_xla_tensors.size()) {
-    return XLA_ERROR_WITH_LOCATION(absl::InvalidArgumentError(
-        absl::StrCat("The size of tensors and new_xla_tensors are not equal: ",
-                     tensors.size(), " vs. ", new_xla_tensors.size())));
+    std::string error_message = absl::StrCat(
+        "Expected the size of the list of tensors (", tensors.size(),
+        ") to match the size of the list of XLATensorPtr (",
```

Comment:
This error message reference C++ details. If this indicate a torch_xla bug, we should crash here.